packages/commonwealth/client/scripts/views/pages/admin_panel/index.tsx

```diff
@@ -0,0 +1,359 @@
+  const [rpcEndpoint, setRpcEndpoint] = React.useState<string>('');
+  const [rpcEndpointChainValueValidated, setRpcEndpointChainValueValidated] =
+    React.useState<boolean>(false);
+  const [rpcEndpointChainNode, setRpcEndpointChainNode] =
+    React.useState<NodeInfo>(null);
+  const [rpcEndpointChainNodeValidated, setRpcEndpointChainNodeValidated] =
+    React.useState<boolean>(false);
+  const [rpcName, setRpcName] = React.useState<string>('');
+  const [bech32, setBech32] = React.useState<string>('');
+  const [balanceType, setBalanceType] = React.useState<BalanceType>(
+    BalanceType.Ethereum
+  );
+  const [chainNodeNotCreated, setChainNodeNotCreated] =
+    React.useState<boolean>(false);
+
+  const buttonEnabled =
+    (rpcEndpointChainNodeValidated &&
+      rpcEndpointChainNode &&
+      rpcEndpointChainValueValidated) ||
+    (rpcName !== '' && bech32 !== '' && rpcEndpoint !== '');
+  return (
+    <div className="TaskGroup">
+      <CWText type="h4">Switch/Add RPC Endpoint</CWText>
+      <CWText type="caption">
+        Changes the RPC endpoint for a specific chain, or adds an endpoint if it
+        doesn't yet exist.
+      </CWText>
+      <div className="MultiRow">
+        <div className="TaskRow">
+          <CWTextInput
+            label="Community Id"
+            value={rpcEndpointChainValue}
+            onInput={(e) => {
+              setRpcEndpointChainValue(e.target.value);
+              if (e.target.value.length === 0)
+                setRpcEndpointChainValueValidated(false);
+            }}
+            inputValidationFn={(value: string) => {
+              const chainInfo = app.config.chains.getById(value);
+              if (!chainInfo) {
+                setRpcEndpointChainValueValidated(false);
+                return ['failure', 'Community not found'];
+              }
+              setRpcEndpointChain(chainInfo);
+              setRpcEndpointChainValueValidated(true);
+              return [];
+            }}
+            placeholder="Enter a community id"
+          />
+          <CWTextInput
+            label="RPC Endpoint"
+            value={rpcEndpoint}
+            onInput={(e) => {
+              setRpcEndpoint(e.target.value);
+            }}
+            inputValidationFn={(value: string) => {
+              if (
+                !detectURL(value) &&
+                value.startsWith('wss://') === false &&
+                value.startsWith('ws://') === false
+              ) {
+                setRpcEndpointChainNodeValidated(false);
+                return ['failure', 'Not a valid URL'];
+              }
+
+              const nodeInfo = app.config.nodes.getByUrl(value);
+
+              if (nodeInfo) {
+                setRpcEndpointChainNode(nodeInfo);
+                setRpcEndpointChainNodeValidated(true);
+              } else {
+                setChainNodeNotCreated(true);
+              }
+
+              return [];
+            }}
+            placeholder="Enter an RPC endpoint"
```

Comment:
A little confused by the update RPC UI. What if we want to update both url and alt_wallet_url, like for Cosmos communities? Also when do we get to update bech32?

Reply:
Interesting. OK. Maybe as a fast follow.